src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -542,18 +632,25 @@ public bool TryGetGuid(out Guid value)
 
             if (_stringHasEscaping)
             {
-                return JsonReaderHelper.TryGetEscapedGuid(span, out value);
-            }
+                int idx = span.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(idx != -1);
 
-            Debug.Assert(span.IndexOf(JsonConstants.BackSlash) == -1);
+                Span<byte> utf8Unescaped = stackalloc byte[span.Length];
```

Comment:
nit: `Debug.Assert` that `span.Length <= JsonConstants.MaximumEscapedGuidLength` to make auditing use of stackalloc easier.